mobile/native/src/unfunded_orders.rs

```diff
@@ -0,0 +1,123 @@
+use futures::FutureExt;
+use std::time::Duration;
+use xxi_node::commons::ChannelOpeningParams;
+
+pub(crate) async fn submit_unfunded_channel_opening_order(
+    funding_address: Address,
+    new_order: NewOrder,
+    coordinator_reserve: u64,
+    trader_reserve: u64,
+    needed_channel_size: u64,
+) -> Result<()> {
+    let node = state::get_node().clone();
+    let bdk_node = node.inner.clone();
+    event::publish(&EventInternal::FundingChannelNotification(
+        FundingChannelTask::Pending,
+    ));
+    let runtime = crate::state::get_or_create_tokio_runtime()?;
+    let (future, remote_handle) = async move {
+        event::publish(&EventInternal::FundingChannelNotification(
+            FundingChannelTask::Pending,
+        ));
+        loop {
+            match bdk_node.get_unspent_txs(&funding_address).await {
+                Ok(ref v) if v.is_empty() => {
+                    tracing::debug!(
+                        address = funding_address.to_string(),
+                        amount = needed_channel_size.to_string(),
+                        "No tx found for address"
+                    );
+                    event::publish(&EventInternal::FundingChannelNotification(
+                        FundingChannelTask::Pending,
+                    ));
+                }
+                Ok(txs) => {
+                    // we sum up the total value in this output and check if it is big enough
+                    // for the order
+                    let total_unspent_amount_received = txs
+                        .into_iter()
+                        .map(|(_, amount)| amount.to_sat())
+                        .sum::<u64>();
+
+                    if total_unspent_amount_received >= needed_channel_size {
+                        tracing::info!(
+                            amount = total_unspent_amount_received.to_string(),
+                            address = funding_address.to_string(),
+                            "Address has been funded enough"
+                        );
+                        break;
+                    }
+                    tracing::debug!(
+                        amount = total_unspent_amount_received.to_string(),
+                        address = funding_address.to_string(),
+                        "Address has not enough funds yet"
+                    );
+                }
+                Err(err) => {
+                    tracing::error!("Could not get utxo for address {err:?}")
+                }
+            }
+            tokio::time::sleep(Duration::from_secs(10)).await;
+        }
+
+        event::publish(&EventInternal::FundingChannelNotification(
+            FundingChannelTask::Funded,
+        ));
+
+        if let Err(error) = bdk_node.sync_on_chain_wallet().await {
+            tracing::error!("Failed at syncing wallet {error:?}")
+        }
+
+        let balance = bdk_node.get_on_chain_balance();
+        tracing::debug!(balance = balance.to_string(), "Wallet synced");
+
```

Comment:
🔧 The second log message can contradict the first one.

Should we just loop until the on-chain balance is enough? We confirmed that the UTXOs are there, but we can also confirm that the balance is up to date.

Reply:
Alternatively, we could try to do it without the manual sync. It might not be necessary if we have broken out of the loop above.

Reply:
I figured syncing the wallet is more expensive than looking up just a single address. But you are right, this would be an alternative. 

> Alternatively, we could try to do it without the manual sync. It might not be necessary if we have broken out of the loop above.

Unfortunately we need to sync the wallet once we found the address holds enough funds as otherwise we might fail later on because the wallet thinks it doesn't have enough funds.

Reply:
> I figured syncing the wallet is more expensive than looking up just a single address. But you are right, this would be an alternative.

Right. My first suggestion was to sync in a loop _after_ the first loop has exited successfully. In theory, you only need to sync once, but there is an error path that we are currently just logging. I think if we fail to sync we want to keep trying until the UTXOs that we think should be there appear in the wallet.

> Unfortunately we need to sync the wallet once we found the address holds enough funds as otherwise we might fail later on because the wallet thinks it doesn't have enough funds.

Okay, so one successful sync is necessary.